klog.go

```diff
@@ -509,7 +509,7 @@ type loggingT struct {
 	addDirHeader bool
 
 	// If set, all output will be redirected unconditionally to the provided logr.Logger
-	logr logr.Logger
+	logr *logr.Logger
```

Comment:
I stumbled over this until I dug into the logr changes and found that logr.Logger is now a struct - then this started to make sense.

A more complete commit message in this PR would have been useful.

Reply:
yep will do @pohly

Reply:
It's not a good idea, a break changed

Reply:
we don't really have a choice here @RyouZhang